qemu/tests/nic_opt.py

```diff
@@ -0,0 +1,149 @@
+            utils_net.get_host_ip_address(params),
+            params.get("server_path", "/var/tmp"),
+            params.get("pkg_md5sum"),
+            os.path.join(data_dir.get_deps_dir("netperf"),
+                         params.get("netperf_server_link")),
+            password=params.get("hostpassword"))
+
+        try:
+            n_server.start()
+            # Run netperf with message size defined in range.
+            netperf_test_duration = int(params.get("netperf_test_duration", 180))
+            netperf_para_sess = params.get("netperf_para_sessions", "1")
+            test_protocols = params.get("test_protocols", "TCP_STREAM")
+            netperf_cmd_prefix = params.get("netperf_cmd_prefix", "")
+            netperf_output_unit = params.get("netperf_output_unit", " ")
+            netperf_package_sizes = params.get("netperf_sizes")
+            test_option = params.get("test_option", "")
+            test_option += " -l %s" % netperf_test_duration
+            if params.get("netperf_remote_cpu") == "yes":
+                test_option += " -C"
+            if params.get("netperf_local_cpu") == "yes":
+                test_option += " -c"
+            if netperf_output_unit in "GMKgmk":
+                test_option += " -f %s" % netperf_output_unit
+            start_time = time.time()
+            stop_time = start_time + netperf_test_duration
+            num = 0
+            for protocol in test_protocols.split():
+                error_context.context("Testing %s protocol" % protocol,
+                                      logging.info)
+                t_option = "%s -t %s" % (test_option, protocol)
+                n_client.bg_start(utils_net.get_host_ip_address(params), t_option,
+                                  netperf_para_sess, netperf_cmd_prefix,
+                                  package_sizes=netperf_package_sizes)
+                if utils_misc.wait_for(n_client.is_netperf_running, 10, 0, 1,
+                                       "Wait netperf test start"):
+                    logging.info("Netperf test start successfully.")
+                else:
+                    test.error("Can not start netperf client.")
+                num += 1
+                # here when set a run flag, when other case call this case as a
+                # subprocess backgroundly,can set this run flag to False to stop
+                # the stress test.
+                env["netperf_run"] = True
+                netperf_test_duration = stop_time - time.time()
+                utils_misc.wait_for(
+                    lambda: not n_client.is_netperf_running(),
+                    netperf_test_duration, 0, 5,
+                    "Wait netperf test finish %ss" % netperf_test_duration)
+                time.sleep(5)
+        finally:
+            n_server.stop()
+            n_server.package.env_cleanup(True)
+            n_client.package.env_cleanup(True)
+            env["netperf_run"] = False
```

Comment:
var in env should be del in finally

Reply:
Thanks,i will delete it in the next version.